### PR TITLE
rbextract: support devices with plain RLE caldata

### PR DESCRIPTION
**Pull request by rogerpueyo**
Very old MikroTik devices have the radio calibration data encoded in RLE and directly stored in the art (i.e., hard_config) partition, without LZO compression nor the preceding ERD magic bytes (which newer devices carry).

This commit adds a fallback for these devices: when no LZO or ERD magic bytes are read, assume raw RLE-encoded calibration data will follow.

Tested on the ath79 target with a MikroTik SXT 5nD r2 (SXT Lite5), which has the calibration data on the SPI NOR. The device is not yet supported though, as the NAND flash is not working yet.

Signed-off-by: Roger Pueyo Centelles <roger.pueyo@guifi.net>

### CODE REVIEW

File: package/utils/rbextract/src/rbextract.c [hunk 1]
```diff
@@ -362,6 +362,18 @@ __rb_get_wlan_data(void)
 
 		buf_rle_out = buf_lzo_out;
 	}
+	/* Even older ath79-base boards directly have RLE-encoded calibration data,
+	without any LZO compresion nor showing RB_MAGIC_ERD bytes */
+	else{
```
Comment (adrianschmutzler) [inline package/utils/rbextract/src/rbextract.c]: missing space

Comment (rogerpueyo) [reply package/utils/rbextract/src/rbextract.c]: D'oh!

**Comment by adrianschmutzler** [comment]
@xback

File: package/utils/rbextract/src/rbextract.c [hunk 2]
```diff
@@ -335,7 +335,7 @@ __rb_get_wlan_data(void)
 	}
 	/* Older ath79-based boards directly show the RB_MAGIC_ERD bytes followed by
 	the LZO-compressed calibration data with no RLE */
-	if (magic == RB_MAGIC_ERD) {
+	else if (magic == RB_MAGIC_ERD) {
```
Comment (adrianschmutzler) [inline package/utils/rbextract/src/rbextract.c]: This needs a little bit of explanation. Is this a fix for an earlier mistake (then it should go into a separate commit) or a desired behavior change?

**Comment by adrianschmutzler** [comment]
Consider adding like "not merged yet" or similar to your Tested-by line, so it obvious to the reader that you tested with a locally supported device.

File: package/utils/rbextract/src/rbextract.c (hunk 2 above)
Comment (rogerpueyo) [reply package/utils/rbextract/src/rbextract.c]: It is a desired behaviour change. The idea is that this last case should only be used if no LZOR and no ERD bytes are detected. Let me explain it in more detail:

Originally, in line282, we checked if the tag began with the RB_MAGIC_LZOR bytes:
```
magic = get_u32(tag);
	if (magic == RB_MAGIC_LZOR) {
		// do stuff
	}
```

Then, in line 338, we checked if the tag began with the RB_MAGIC_ERD bytes:
```
 	if (magic == RB_MAGIC_ERD) { 
 	 	// do other stuff
 	}
```

Both cases were mutually exclusive. Now, however, for these even older devices  _**without**_ RB_MAGIC_ERD _**nor**_ RB_MAGIC_LZOR bytes, we need to resort to the third mutually exclusive case I've added. The thing is that RLE does not have "magic bytes", so I had to change line 338 so that each type of device falls only into one of the cases (otherwise, RB_MAGIC_LZOR devices would also get into the new third case).

Maybe a switch statement would be more elegant?
```
magic = get_u32(tag);
switch (magic) {
	case RB_MAGIC_LZOR:
		// do stuff
		break;
	case RB_MAGIC_ERD:
		// do other stuff
		break;
	default:
		// do some other stuff
}
```

Comment (adrianschmutzler) [reply package/utils/rbextract/src/rbextract.c]: Ah, I haven't looked well enough: Without the "else" added in L365, the if/elseif change would have been cosmetical (I wasn't aware of that before.) Code flow in the new version is obvious, I wouldn't go for a switch with such large code blocks.
So, I'm fine, and somebody has to actually review the content.

**Comment by xback** [comment]
lgtm

**Comment by xback** [comment]
Merged.
Thanks!
